packages/dds/tree/src/feature-libraries/schema-index/schemaSummarizer.ts

```diff
@@ -51,23 +51,26 @@ export class SchemaSummarizer implements Summarizable {
 		});
 	}
 
-	public getAttachSummary(
-		stringify: SummaryElementStringifier,
-		fullTree?: boolean,
-		trackState?: boolean,
-		telemetryContext?: ITelemetryContext,
-		incrementalSummaryContext?: IExperimentalIncrementalSummaryContext | undefined,
-	): ISummaryTreeWithStats {
+	public summarize(props: {
+		stringify: SummaryElementStringifier;
+		fullTree: boolean;
+		trackState?: boolean;
+		telemetryContext?: ITelemetryContext;
+		incrementalSummaryContext?: IExperimentalIncrementalSummaryContext;
+	}): ISummaryTreeWithStats {
+		const incrementalSummaryContext = props.incrementalSummaryContext;
 		const builder = new SummaryTreeBuilder();
+		const fullTree = props.fullTree ?? false;
 		if (
+			!fullTree &&
 			incrementalSummaryContext !== undefined &&
```

Comment:
The fact that we don't do an incremental summary if either fullTree or incrementalSummaryContext is undefined makes me wonder:

do we need both of these parameters? Can we just remove fullTree, and ensure incrementalSummaryContext is set to undefined when a full tree summary is required? At least for how its used in SharedTree, that seems like it would work, and avoid the possibility of accidently not obeying the fullTree request.

Reply:
I think we could merge them in this PR at the SharedObjectKernel level.

Unless I'm missing something, this can be done by simply discarding the incrementalSummaryContext if fullTree is true and not passing in fullTree right?

Since everything downstream from that can be adjusted accordingly as a non-breaking change (it's all internal), it seems like we can just do that now while we are updating all the involved methods anyway.

That said, this PR is still progress, so I'm going to approve it for now.

Reply:
We can't discard `incrementalSummaryContext` even if fullTree is true. It contains sequence numbers of the current summary and the last one which DDS may need to track even if a particular summary isn't incremental.

The other approach is to set `fullTree` to true if `incrementalSummaryContext` is undefined. But that means DDS will still have to check both before usage so it's not particularly useful.
We could update the types such that if `fullTree` is false or undefined then `incrementalSummaryContext` must be defined and if `fullTree` is true, `incrementalSummaryContext` may be defined. I will do that in a separate PR since I am struggling a bit with typing and I would like to get this PR in.